web/modules/custom/node_view/node_view.module

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/**
+ * @file
+ * Primary module hooks for node_view_counter module.
+ */
+
+/**
+ * Function to increase count of visited of a node page and display it as well.
+ *
+ * @param array $build
+ * @param \Drupal\Core\Entity\EntityInterface $entity
+ * @param \Drupal\Core\Entity\Display\EntityViewDisplayInterface $display
+ * @return void
+ */
+function node_view_entity_view(array &$build, \Drupal\Core\Entity\EntityInterface $entity, \Drupal\Core\Entity\Display\EntityViewDisplayInterface $display)
+{
+  // Check if the entity is a node.
+  if ($entity->getEntityTypeId() === 'node') {
+    // Get the current session.
+    $session = \Drupal::request()->getSession();
+    // Get the node ID.
+    $nid = $entity->id();
+    // Get the current count from the session.
+    $count = $session->get('node_view_count_' . $nid, 0);
+    // Increment the count.
+    $count++;
+    // Store the updated count back in the session.
+    $session->set('node_view_count_' . $nid, $count);
+    // Add the count to the node render array.
+    $build['view_count'] = [
+      '#markup' => '<p>This node has been viewed ' . $count . ' times in this session.</p>',
+      '#weight' => 100,
+      '#cache' => [
+        'contexts' => ['session'],
+        'max-age' => 0,
```

Comment:
Let's not use 
'max-age' => 0,

Please update implementation. Ideally we shouldn't be using 'max-age' => 0 at all

Reply:
It's great that now we have removed 
`'max-age' => 0,`
But what other alternative we are using here ?

Reply:
That is incorrect. 
We cannot just disable caching for the entire website just because of our requirement.
We need to invalidate cache here. 
Please find some other solution